ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -904,3 +910,101 @@ def test_ipahealthcheck_trust_agent_member_check(self):
+    reports correct status when IPA server is configured
+    with external CA.
+    """
+
+    @classmethod
+    def install(cls, mh):
+        result = install_server_external_ca_step1(cls.master)
+        assert result.returncode == 0
+        root_ca_fname, ipa_ca_fname = tasks.sign_ca_and_transport(
+            cls.master, paths.ROOT_IPA_CSR, ROOT_CA, IPA_CA
+        )
+
+        install_server_external_ca_step2(
+            cls.master, ipa_ca_fname, root_ca_fname
+        )
+        tasks.kinit_admin(cls.master)
+
+    def test_ipahealthcheck_crlmanagercheck(self):
+        """
+        Test for IPACRLManagerCheck
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.roles", "IPACRLManagerCheck"
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] is True
+
+    def test_ipahealthcheck_certmongerca(self):
+        """
+        Test for IPACertmongerCA
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPACertmongerCA",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert (
+                check["kw"]["key"] == "IPA"
+                or check["kw"]["key"] == "dogtag-ipa-ca-renew-agent"
+                or check["kw"]["key"] == "dogtag-ipa-ca-renew-agent-reuse"
+            )
+
+    def test_ipahealthcheck_ipaopensslchainvalidation(self):
+        """
+        Test for IPAOpenSSLChainValidation
+        """
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPAOpenSSLChainValidation",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert (
+                check["kw"]["key"] == paths.HTTPD_CERT_FILE
+                or check["kw"]["key"] == paths.RA_AGENT_PEM
+            )
+
+    def test_ipahealthcheck_ipansschainvalidation(self):
+        """
+        Test for IPANSSChainValidation
+        """
+        instance = realm_to_serverid(self.master.domain.realm)
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPANSSChainValidation",
+        )
+        assert returncode == 0
+        for check in data:
+            if check["kw"]["dbdir"] == (paths.PKI_TOMCAT_ALIAS_DIR):
+                assert check["result"] == "SUCCESS"
+                assert check["kw"]["nickname"] == "Server-Cert cert-pki-ca"
+            elif check["kw"]["dbdir"] == (
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+            ):
+                assert check["result"] == "SUCCESS"
+                assert check["kw"]["nickname"] == "Server-Cert"
```

Comment:
To test failures you could, for example, delete the IPA cert from the database.

I suspect running ipa-certupdate afterwards and running ipactl restart in a fixture would restore things. NSS doesn't often like databases changing while they are running.

Reply:
Removed the existing test and tried deleting the 'dogtag-ipa-ca-renew-agent' cert using getcert and an error msg is displayed 'Certmonger CA '{key}' missing'